### PR TITLE
Notarization tweaks (take 2)

### PR DESCRIPTION
**Pull request by cadsit**
Issues with the last PR:
- pkgs.flatten() not approved (approved between of https://jenkins.internalx.com/view/All/job/xamarin-build-tasks-master/324/console and https://jenkins.internalx.com/view/All/job/xamarin-build-tasks-master/325/console)
- Uploading an empty glob fails in WAStoragePlugin

### CODE REVIEW

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -579,7 +574,10 @@ timestamps {
 
                             sh ("ls -la package")
                             uploadFiles ("package/*", "wrench", virtualPath)
-                            uploadFiles ("package/notarized/*", "wrench", virtualPath)
+
+                            def notarized_files = findFiles (glob: "package/notarized/*")
+                            if (notarized_files.size () > 0)
+                              uploadFiles ("package/notarized/*", "wrench", virtualPath)
```
Comment (mandel-macaque) [inline jenkins/Jenkinsfile]: Would we like some kind of warning here? As in, we expected to find files, not were found ⚠️ ⚠️ ? Not that it block the PR.

File: jenkins/Jenkinsfile [hunk 2]
```diff
@@ -503,28 +501,26 @@ timestamps {
...
-                                    if (notarize_ios)
-                                        pkgs = pkgs + xiPackages
                                     withCredentials([string(credentialsId: 'codesign_keychain_pw', variable: 'KEYCHAIN_PASS'), string(credentialsId: 'team_id', variable: 'TEAM_ID'), string(credentialsId: 'application_id', variable: 'APP_ID'), string(credentialsId: 'installer_id', variable: 'INSTALL_ID'), usernamePassword(credentialsId: 'apple_account', passwordVariable: 'APPLE_PASS', usernameVariable: 'APPLE_ACCOUNT')]) {
                                         sh (returnStatus: true, script: "security create-keychain -p ${env.KEYCHAIN_PASS} login.keychain") // needed to repopulate the keychain
                                         sh ("security unlock-keychain -p ${env.KEYCHAIN_PASS} login.keychain")
-                                        sh ("python release-scripts/sign_and_notarize.py -a ${env.APP_ID} -i ${env.INSTALL_ID} -u ${env.APPLE_ACCOUNT} -p ${env.APPLE_PASS} -t ${env.TEAM_ID} -d package/notarized -e ${entitlements} -k login.keychain " + pkgs.flatten().join(" "))
+                                        timeout(time: 90, unit: 'MINUTES') {
```
Comment (mandel-macaque) [inline jenkins/Jenkinsfile]: As with the other PR, lets me consistent.
```suggestion
                                        timeout (time: 90, unit: 'MINUTES') {
```

File: jenkins/Jenkinsfile (hunk 1 above)
Comment (cadsit) [reply jenkins/Jenkinsfile]: @mandel-macaque Should be partially covered by https://github.com/xamarin/xamarin-macios/blob/04723c517f81a3344411d44e0750d7b57e66aadb/jenkins/Jenkinsfile#L536-L542. I can add in more warnings there (for when notarization succeeds and pkgs aren't where we think they are).

**Comment by mandel-macaque** [comment]
Known crash: https://github.com/xamarin/maccore/issues/581

**Comment by rolfbjarne** [comment]
Test failures are unrelated

* monotouch-test/tvOS - simulator/Debug (all optimizations): https://github.com/xamarin/maccore/issues/2040
* monotouch-test/watchOS 32-bits - simulator/Debug: https://github.com/xamarin/maccore/issues/581
* [NUnit] Mono BCL tests group 2/tvOS - device/AssemblyBuildTarget: dylib (debug, profiling): https://github.com/xamarin/maccore/issues/1124

**Pull request merged**